src/io/iter_image_recordio_2.cc

```diff
@@ -518,6 +518,17 @@ inline unsigned ImageRecordIOParser2<DType>::ParseChunk(DType* data_dptr, real_t
       cv::Mat res;
       rec.Load(blob.dptr, blob.size);
       cv::Mat buf(1, rec.content_size, CV_8U, rec.content);
+
+      if (idx % 1000 == 0) {
+        if (param_.seed_aug.has_value()) {
+          LOG(INFO) << "aug seed: " << param_.seed_aug.value();
+        }
+        LOG(INFO) << "tid: " << tid << " idx: " << idx << " index: " << rec.image_index();
+      }
+      if (param_.seed_aug.has_value()) {
+        prnds_[tid]->seed(idx + param_.seed_aug.value());
```

Comment:
I do not understand why seeds need to be reset all the time. The logic is a bit obscure - the random generators are in the augmentor and get passed to the iterator. 
Anyway, what you could try to do (if not refactoring) is to seed all generators in `DefaultImageAugmenter::Init` [here](https://github.com/apache/incubator-mxnet/blob/master/src/io/image_aug_default.cc#L215) or in a separate method (since the random generator is not passed in to `Init`). I guess the interface `ImageAugmenter` would require a separate `seed` method then that could be called sequentially (instead of openmp parallelized).

Reply:
There are two requirements here: that setting the seed will yield reproducible results and that parallelization should be used to augment the images. We need to reset the seed for each image because there is no guarantee that the same image will be processed by the same thread. Or that even that it will be the ith image processed by that thread across every run (or even different hardware - the code figures out the number of threads to use for processing). Therefore, resetting the random number generator at the start of processing an image it the only way (at least that I could think of) to guarantee that, in the case of setting a fixed seed, the same random distortions will be applied to the same image, in a multi-threaded environment, independent of the hardware being used. I hope this is clear. It's not the easiest topic to discuss in written form lol.